frontend/packages/core-api-client/src/lib/BaseRESTClient.ts

```diff
@@ -1,31 +1,40 @@
-import axios, { AxiosInstance, Method } from 'axios';
+import axios, { AxiosInstance, AxiosRequestConfig, Method } from 'axios';
 
 import { RequestOptions, Response } from './types';
 import { clientResponse } from './utils/responses';
 
 export class BaseRESTClient {
   protected readonly axiosInstance: AxiosInstance;
 
+  lastInterceptorId: number | undefined;
+
   constructor(baseURL: string) {
     this.axiosInstance = axios.create({
       baseURL,
     });
   }
 
   public setAuth = (token: string): void => {
-    this.axiosInstance.interceptors.request.use((value: any) => {
-      const result = { ...value };
-      if (!token) {
-        return value;
-      }
-      return {
-        ...result,
-        headers: {
-          ...result.headers,
+    console.log('Setting token to:', token);
+
+    if (this.lastInterceptorId != null)
```

Comment:
I think rather than ejecting the interceptor and using a new one, we could do a slight refactor. 

* Move the interceptor logic into the constructor
* Rather than taking `token` as a function argument, it's a method on the class `this.getToken()`. This will be called on each intercept.
* Add another class method `setToken` that will just store it as a property.
* When using `AuthWrapper` use `setToken` instead of `setAuth`

Reply:
This package also seems useful, but I'm not sure it fits in with how we've structured auth. Potentially something for a refactor later down the line.
https://www.npmjs.com/package/axios-auth-refresh